indexing-service/src/test/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisorStateTest.java

```diff
@@ -1214,16 +1248,19 @@ public LagStats computeLagStats()
     private final CountDownLatch latch;
     private final Map<String, Long> partitionsRecordLag;
     private final Map<String, Long> partitionsTimeLag;
+    private boolean emitNoticeQueueSize;
 
     TestEmittingTestSeekableStreamSupervisor(
         CountDownLatch latch,
         Map<String, Long> partitionsRecordLag,
-        Map<String, Long> partitionsTimeLag
+        Map<String, Long> partitionsTimeLag,
+        boolean emitNoticeQueueSize
```

Comment:
any reason we need this flag at all?

Reply:
Because I didn't want to mix up the tests that are testing the emitlag logic by emitting noticesQueue size metric. So explicitly disabling it in all lag related tests and added a separate test for this new metric.

Reply:
it's a little awkward to have this flag. I think it would nicer to update the existing tests to ignore metrics unrelated to what they are testing. This will make it easier to add new metrics in the future.